components/datetime/src/format/time_zone.rs

```diff
@@ -100,27 +91,37 @@ where
     /// time_zone.time_zone_id = Some(tinystr!(8, "zzzzz").into());
     /// assert!(matches!(
     ///     tzf.format(&time_zone).write_no_fallback(&mut buf),
-    ///     Err(DateTimeError::UnsupportedOptions)
+    ///     Ok(Err(DateTimeWriteError::MissingTimeZoneSymbol(_)))
     /// ));
     ///
     /// // Use the `Writable` trait instead to enable infallible formatting:
     /// writeable::assert_writeable_eq!(tzf.format(&time_zone), "GMT");
     /// ```
-    pub fn write_no_fallback<W>(&self, mut w: &mut W) -> Result<fmt::Result, DateTimeError>
+    pub fn write_no_fallback<W>(
+        &self,
+        mut w: &mut W,
```

Comment:
Are we keeping `write_no_fallback`? I assumed that with `TryWriteable` we would only have `write` which does the "last resort" fallback for the lossy path.

Reply:
I intend for it to go away, but for now this change doesn't impact tests / APIs